### ISSUE TITLE
[Edit] Importing settings from a .json file does not work

### ISSUE THREAD

**Issue by Korb** [erosman/support#629]
I exported the add-on settings to a json file, edited it, and tried importing via Options / Import. However, after selecting a file and confirming the selection, nothing happens. I assumed that when importing, the presence of added records is checked with existing ones, and matching ones are not added, and I removed all custom start & end strings from the add-on. However, even after this the import was not successful, without any error messages.

Mozilla Firefox 125.0b1 (64-bit)
Edit 1.9 (January 17, 2023)
Microsoft Windows 10.0.19045.4170

**Comment by erosman** [comment]
Did you click SAVE after importing?

**Comment by Korb** [comment]
No, it never even occurred to me to do that. I didn’t see any changes that I would like to _save_.

I have now checked and made sure that the changes begin to be displayed only after clicking on the "Save" button.

Please add a hint that the imported settings will only be applied after clicking the "Save" button. This is not obvious now. For most user applications, selecting a file is enough to start processing it.

**Comment by erosman** [comment]
> Please add a hint that the imported settings will only be applied after clicking the "Save" button. 

It is in the Help.

![image](https://github.com/erosman/support/assets/6458876/9b4fac6c-83d4-4109-b7f7-7c4dbaf5a69c)

**Comment by Korb** [comment]
> Preferences: Import/Export
> You can import/export Preferences (for backup or share) from/to a local file on your computer.
> Import is non-destructive. Click save to apply the new settings.
> Save
> Please note that changes will not take effect until they are saved.

I exported my current custom tags settings, reordered the tags using Notepad++

[Edit_2024-05-31.json](https://github.com/user-attachments/files/15510323/Edit_2024-05-31.json)

and tried importing without success. Mozilla Firefox 127.0b8 gave a blank notification

![2024-05-31_11-02-43](https://github.com/erosman/support/assets/3509797/c12bd078-fc0a-412a-94ff-6a52813141a0)

and the list of custom tags remained the same. I imported empty json

```
{
  "custom": []
}
```

The list of tags in the add-on has predictably been cleared. But even then my modified json with a set of my tags refused to be imported.

Edit 1.9 (January 17, 2023)
Microsoft Windows 10.0.19045.4474

**Comment by erosman** [comment]
> I exported my current custom tags settings, reordered the tags using Notepad++
> 
> [Edit_2024-05-31.json](https://github.com/user-attachments/files/15510323/Edit_2024-05-31.json)

The JSON is invalid. 
You can test it on [JSONLint](https://jsonlint.com/).

**Comment by Korb** [comment]
I understand that advising on JSON syntax issues is not at all the scope of this bug tracker. However, I would not bother exporting and importing settings, just like editing exported files, if the add-on had the ability to arbitrarily change the list of tags in groups, or at least the sorting of custom tags worked stably, preferably by dragging and dropping. (Currently, the up and down buttons most often do not work, even after saving the list of custom tags.) So, I'm trying to edit the list of custom tags using a word processor, ~~and I don't understand why the same entry can be correct and incorrect at the same time~~ . I didn't know that in JSON the last entry must end with `}`, but not `},` like all the others. Thank you!

**Comment by erosman** [comment]
> I didn't know that in JSON the last entry must end with }, but not }, like all the others. 

JSON has a very strict format. [Trailing commas](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Trailing_commas) are acceptable in JavaScript but not in JSON.

>  Is it possible to refuse to display a blank browser notification?

Not at the moment, but it can be a feature request.
